core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -145,6 +145,30 @@ private ShareSessionKey shareSessionKey(String groupId, Uuid memberId) {
         return new ShareSessionKey(groupId, memberId);
     }
 
+    public Errors shareAcknowledgeError(String groupId, Uuid memberId, int reqEpoch) {
+        if (reqEpoch == ShareFetchMetadata.INITIAL_EPOCH)
+            return Errors.INVALID_SHARE_SESSION_EPOCH;
+        else {
+            ShareSessionKey key = shareSessionKey(groupId, memberId);
+            ShareSession shareSession = cache.get(key);
+            if (shareSession == null) {
+                log.debug("Share session error for {}: no such share session found", key);
+                return Errors.SHARE_SESSION_NOT_FOUND;
+            }
+            if (reqEpoch == ShareFetchMetadata.FINAL_EPOCH) {
+                if (cache.remove(key) != null)
```

Comment:
This seems like quite a significant side-effect in an error-mapping method. I'd prefer the cache removal to be outside this method.

Reply:
makes sense, I'll handle it in a different function